boofuzz/primitives/numbers.py

```diff
@@ -0,0 +1,150 @@
+import six
+
+from .base_primitive import BasePrimitive
+from .. import helpers
+
+
+class Numbers(BasePrimitive):
+    def __init__(self,
+        value,
+        size=-1,
+        max_num=None,
+        signed=False,
+        fuzzable=True,
+        name=None,
+    ):
+        """
+        Primitive that cycles through a library of "bad" numbers, represented as strings.
+
+        @type  value:         str
+        @param value:         Default number value
+        @type  size:          int
+        @param size:          (Optional, def=-1) Static size of this field, leave -1 for dynamic.
```

Comment:
This seems to be used for padding only. Truncating should be possible too.

Reply:
Truncating them could work, though it wouldn't make much sense IMO.
I left these params so that they could be used in the future. Didn't want to decide on how to use them...

Reply:
Imagine a case where this field is somewhere in the middle of the frame and because it is not truncated correctly everything behind it gets shifted backwards. This would corrupt the whole structure of the frame. Sometimes a field has to have a fixed size.

If it doesn't make sense to truncate values they should be filtered out, so that only values with the same length or shorter (with padding) are added.